src/accountsdb/db.zig

```diff
@@ -2254,35 +2708,121 @@ fn testWriteSnapshotFull(
     const snap_fields = try SnapshotFields.decodeFromBincode(allocator, manifest_file.reader());
     defer snap_fields.deinit(allocator);
 
-    const status_cache_file = try snapshot_dir.openFile("snapshots/status_cache", .{});
-    defer status_cache_file.close();
+    try accounts_db.loadFromSnapshot(snap_fields.accounts_db_fields, 1, allocator);
 
-    const status_cache = try StatusCache.decodeFromBincode(allocator, status_cache_file.reader());
-    defer status_cache.deinit(allocator);
+    var tmp_dir_root = std.testing.tmpDir(.{});
+    defer tmp_dir_root.cleanup();
+    const tmp_dir = tmp_dir_root.dir;
 
-    var accounts_db = try AccountsDB.init(allocator, .noop, snapshot_dir, .{
-        .number_of_index_bins = ACCOUNT_INDEX_BINS,
-        .use_disk_index = false,
-    });
-    defer accounts_db.deinit(true);
+    const archive_file = blk: {
+        var snapshot_gen_pkg, const snapshot_gen_info = try accounts_db.makeFullSnapshotGenerationPackage(
+            slot,
+            snap_fields.bank_fields,
+            snap_fields.lamports_per_signature,
+            snap_fields.accounts_db_fields.stored_meta_write_version,
+        );
+        defer snapshot_gen_pkg.deinit();
+
+        try std.testing.expectEqual(slot, snapshot_gen_info.slot);
+        if (maybe_expected_hash) |expected_hash| {
+            try std.testing.expectEqual(expected_hash, snapshot_gen_info.hash);
+        }
+
+        const archive_file_name_bounded = sig.accounts_db.snapshots.FullSnapshotFileInfo.snapshotNameStr(.{
+            .slot = snapshot_gen_info.slot,
+            .hash = snapshot_gen_info.hash,
+            .compression = .zstd,
+        });
+        const archive_file_name = archive_file_name_bounded.constSlice();
+        const archive_file = try tmp_dir.createFile(archive_file_name, .{ .read = true });
+        errdefer archive_file.close();
 
-    try accounts_db.loadFromSnapshot(snap_fields.accounts_db_fields.file_map, 1, allocator);
+        var buffered_state = std.io.bufferedWriter(archive_file.writer());
+        try snapshot_gen_pkg.write(buffered_state.writer());
+        try buffered_state.flush();
+
+        try accounts_db.commitFullSnapshotInfo(snapshot_gen_info, .ignore_old);
+
+        break :blk archive_file;
+    };
+    defer archive_file.close();
+
+    var actual_snapshot_dir = try tmp_dir.makeOpenPath("output", .{ .iterate = true });
+    defer actual_snapshot_dir.close();
+
+    try archive_file.seekTo(0);
+    try std.tar.pipeToFileSystem(actual_snapshot_dir, archive_file.reader(), .{});
+
+    {
```

Comment:
we should probably also include validating from the snapshot in the unit tests too

Reply:
we'll have to leave this as future work for when we revive the unit test.

Reply:
68730af34847e389077fa52359b4a4b3f6b0f34f